### ISSUE TITLE
Determinism

### ISSUE THREAD

**Issue by jcaudle** [bigmachine-io/imposter-v2#2]
On page 11, you equate determinism with the idea that everything has a cause. Technically speaking (in philosophy), determinism means something very different: that knowing the present state of the world will allow you to know what the next state will be. I would probably just remove the reference to determinism since I also don’t know how it’s related to Clarke’s three rules, but that’s just me.

I hope this is the right kind of technical note, but if it’s better to give as general feedback by email, I’m happy to switch to that. Thanks! 🙂

**Comment by robconery** [comment]
Nope - this is fine as a venue. Do you have a reference I can chase down RE determinism and philosophy? I've never seen the term used with future events - only probability. I very much _do not_ want to remove the reference as it underpins the entire idea of known vs. unknown, logic and deduction vs. probability and inference. See what I mean?

**Comment by jcaudle** [comment]
The [Stanford Encyclopedia of Philosophy entry on _Causal Determinism_](https://plato.stanford.edu/entries/determinism-causal/) is quite good. [The introduction to the entry](https://plato.stanford.edu/entries/determinism-causal/#Int) is where the distinction I made in the original post is drawn out, though the abstract at the top of the page seems to support the way you described determinism in the book.

I do see how the concepts are related on a second reading, but I think the section (and the later reference to a deterministic past) would be a little tighter without the reference to the term `determinism` so much as the concept the term represents. I like `Cause and Effect forest` as a descriptor, but I don't know that `determinism` is the right technical term for that idea.

**Comment by robconery** [comment]
Would it be fair to say your issue is between `deterministic` and `determinism`? I'm trying to understand why you want me to tighten/remove the term. Maybe if I explained my thinking a bit more we could figure something out.

Clarke's 3 Laws are essentially the notion that if you stop looking, whatever lies beyond the horizon loses its cause and therefore can be ascribed to magic. Hanselman's example is his wife's knowledge of the sink drain - beyond the drain is mystery.

All that's needed to uncover the cause/effect crumb trail is effort. You can _determine_ how something happens by diving in and digging up the details. You can draw out a binary decision tree to describe exactly how a given event has occurred, _if you choose to dig deep/long enough_. We know this because (for now) because no one has proven P = NP. Once that happens, all bets are off :).

This is my thinking and I'm happy to tweak... but I need more than a vague sense of wrongness. I read the entry you linked and I feel comfortable with it and my interpretation - but I could be completely wrong.

**Comment by jcaudle** [comment]
> Would it be fair to say your issue is between `deterministic` and `determinism`?

Definitely. I'm probably just getting hung up on this since I spent a significant time in college studying precisely this question and human action.

Maybe I'd say that things in the past were `determined` already and that if you look backward you can determine what that cause and effect relationship was. The problem I have with saying that this is `determinism` is that it connotes a determined future too. I like the direction you're going with this though.

Feel free to close the issue unless you have more questions! I'm looking forward to seeing the tweaks keep coming!

**Comment by robconery** [comment]
Will do - I'd like to keep this open as it's fascinating! 

>The problem I have with saying that this is determinism is that it connotes a determined future too.

Eek - I'm trying to make the exact opposite point. Hmmm - maybe there's a tweak I can put in to make that obvious?

**Comment by jcaudle** [comment]
It may be too technical, but I think [the principle of sufficient reason](https://en.wikipedia.org/wiki/Principle_of_sufficient_reason) is what you’re getting at. It’s certainly more of a mouthful, but it dodges all of the connotations around the future being determined and only insists on the past being full of causes.
